api/token.go

```diff
@@ -150,13 +150,16 @@ func (a *API) ResourceOwnerPasswordGrant(ctx context.Context, w http.ResponseWri
 		return unprocessableEntityError("Only an email address or phone number should be provided on login.")
 	}
 	var user *models.User
+	var provider string
 	var err error
 	if params.Email != "" {
+		provider = "email"
```

Comment:
An initial thought was that we could change this into an Enum. However, I later realized that this might not make sense given the number of providers we have.

Reply:
Hmm that's a good idea though, we can refactor this into an enum to eliminate accidental typo errors in a separate PR.
Something like the following:

```go
type ProviderName string

const (
  Email ProviderName = "email"
  Google ProviderName = "google"
  Facebook ProviderName = "facebook"
   ...
)
```